cpp-fs/storage/storage/storage.cpp

```diff
@@ -0,0 +1,20 @@
+#include "storage.hpp"
+
+#include <boost/uuid/string_generator.hpp>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+namespace cppfs::storage {
+
+/// This function is a modification of the function from
+/// https://stackoverflow.com/a/28685177
+bool IsUUIDValid(std::string const& str) noexcept {
+  using namespace boost::uuids;
+  try {
+    return string_generator()(str).version() != uuid::version_unknown;
+  } catch (...) {
+    return false;
+  }
```

Comment:
Why do we suppress errors here?

Reply:
This function may throw an exception if you pass an incorrect string. It is assumed that this string can also be specified by the user, so we must be able to handle such a case.
If an exception is thrown, it means that an incorrect identifier was passed.

Reply:
Sounds like we only want to catch this specific type of exception here